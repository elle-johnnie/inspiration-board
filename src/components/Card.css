@import url('https://fonts.googleapis.com/css?family=Merriweather+Sans|Shadows+Into+Light');

.card {
  background-color: aqua;

  padding: 1em 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns: 1fr 5fr 1fr;
  align-items: center;
}

.card__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: hidden;
    font-size: 2.5rem;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
  font-family: 'Shadows Into Light', cursive;

}

.card__delete {
  color: orangered;
  font-family: 'Merriweather Sans', sans-serif;}

.card__update {
  color: darkgreen;
  font-family: 'Merriweather Sans', sans-serif;}

.buttons{
  display: flex;
  padding-top: 3rem;
  justify-content: space-around;


}

/* Curl Bottom Right */
.hvr-curl-bottom-right {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  position: relative;
}
.hvr-curl-bottom-right:before {
  pointer-events: none;
  position: absolute;
  content: '';
  height: 0;
  width: 0;
  bottom: 0;
  right: 0;
  background: white;
  /* IE9 */
  background: linear-gradient(315deg, white 45%, #aaa 50%, #ccc 56%, white 80%);
  box-shadow: -1px -1px 1px rgba(0, 0, 0, 0.4);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: width, height;
  transition-property: width, height;
}
.hvr-curl-bottom-right:hover:before, .hvr-curl-bottom-right:focus:before, .hvr-curl-bottom-right:active:before {
  width: 25px;
  height: 25px;
}

/* Grow Shadow */
.hvr-grow-shadow {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow, transform;
  transition-property: box-shadow, transform;
}
.hvr-grow-shadow:hover, .hvr-grow-shadow:focus, .hvr-grow-shadow:active {
  box-shadow: 0 10px 10px -10px rgba(0, 0, 0, 0.5);
  -webkit-transform: scale(1.1);
  transform: scale(1.1);
}
