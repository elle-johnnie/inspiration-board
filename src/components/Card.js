import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
    constructor(props) {
        super(props);
        this.state = {
            update: false,
        }
    };

    handleDelete = () => {
        console.log(this.props.id);
        this.props.removeCardCallback(this.props.id)
    };

    handleUpdate = () => {
        console.log(this.props.id);
        this.props.updateCardCallback(this.props.id)
    };


    render() {
        const {id, text} = this.props;
        let icon = this.props.emoji;
        if (icon !== undefined) {
            icon = emoji.getUnicode(icon);
        }
        return (
            <div className="card hvr-curl-bottom-right" key={id}>
                <section className="card__content">
                    <p className="card__content-text"> {text} </p>
                    <p className="card__content-emoji"> {icon} </p>
                    <section className="buttons">
                    {/*<button*/}
                        {/*onClick={this.handleUpdate}*/}
                        {/*type="button"*/}
                        {/*className="card__update btn hvr-grow-shadow"*/}
                        {/*aria-label="Update"*/}
                    {/*> UPDATE*/}
                    {/*</button>*/}
                    <button
                        onClick={this.handleDelete}
                        type="button"
                        className="card__delete btn hvr-grow-shadow"
                        aria-label="Remove"
                    > REMOVE
                    </button>
                    </section>
                </section>
            </div>
        );
    }
}

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string,
  emoji: PropTypes.string,
  // updateCardCallback: PropTypes.func,
  removeCardCallback: PropTypes.func,

};

export default Card;
