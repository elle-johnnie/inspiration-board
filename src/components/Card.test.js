import {shallow} from "enzyme/build";
import React from "react";
import Card from './Card';

describe('Card', () => {
    test('that it matches an existing snapshot', () => {
        // First Mount the Component in the testing DOM
        // Arrange
        // use proptypes to determine list of req'd props
        const wrapper = shallow(<Card
            text='you are amazing'
            id={1}
            emoji="heart"
            removeCardCallback={() => { }}
        />);

        // Assert that it looks like the last snapshot
        expect(wrapper).toMatchSnapshot();
    });

    test('card renders when no emoji', () => {
        const wrapper = shallow( <Card
            text='you are better than amazing'
            id={2}
            emoji=""
            removeCardCallback={() => { }}
        />);
        expect(wrapper).toMatchSnapshot();
    });

    test('card renders when no text', () => {
        const wrapper = shallow( <Card
            text='you are better than everything amazing'
            id={3}
            emoji="heart"
            removeCardCallback={() => { }}
        />);
        expect(wrapper).toMatchSnapshot();
    });
});