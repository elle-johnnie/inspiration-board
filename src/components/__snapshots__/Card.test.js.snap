// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card card renders when no emoji 1`] = `
<div
  className="card hvr-curl-bottom-right"
  key="2"
>
  <section
    className="card__content"
  >
    <p
      className="card__content-text"
    >
       
      you are better than amazing
       
    </p>
    <p
      className="card__content-emoji"
    >
       
       
    </p>
    <section
      className="buttons"
    >
      <button
        aria-label="Remove"
        className="card__delete btn hvr-grow-shadow"
        onClick={[Function]}
        type="button"
      >
         REMOVE
      </button>
    </section>
  </section>
</div>
`;

exports[`Card card renders when no text 1`] = `
<div
  className="card hvr-curl-bottom-right"
  key="3"
>
  <section
    className="card__content"
  >
    <p
      className="card__content-text"
    >
       
      you are better than everything amazing
       
    </p>
    <p
      className="card__content-emoji"
    >
       
      ❤️
       
    </p>
    <section
      className="buttons"
    >
      <button
        aria-label="Remove"
        className="card__delete btn hvr-grow-shadow"
        onClick={[Function]}
        type="button"
      >
         REMOVE
      </button>
    </section>
  </section>
</div>
`;

exports[`Card that it matches an existing snapshot 1`] = `
<div
  className="card hvr-curl-bottom-right"
  key="1"
>
  <section
    className="card__content"
  >
    <p
      className="card__content-text"
    >
       
      you are amazing
       
    </p>
    <p
      className="card__content-emoji"
    >
       
      ❤️
       
    </p>
    <section
      className="buttons"
    >
      <button
        aria-label="Remove"
        className="card__delete btn hvr-grow-shadow"
        onClick={[Function]}
        type="button"
      >
         REMOVE
      </button>
    </section>
  </section>
</div>
`;
